scripts/deploy-federation.sh

```diff
@@ -117,11 +117,13 @@ else
 fi
 
 # Create a permissive clusterrolebinding to allow federation controllers to run.
-# TODO(marun) Make this more restrictive.
-# TODO(marun) Use a rolebinding when namespaced once the
-# federaatedtypeconfig controller can limit informer scope to the
-# namespace.
-kubectl create clusterrolebinding federation-admin --clusterrole=cluster-admin --serviceaccount="${NS}:default"
+if [[ "${NAMESPACED}" ]]; then
+  # TODO(marun) Investigate why cluster-admin is required to view cluster registry clusters in a namespace
```

Comment:
Should cluster registry clusters be labeled as `system:aggregate-to-admin` or `system:aggregate-to-edit`?  It looks like label selection is [not supported](https://github.com/kubernetes/kubernetes/issues/44703) in rbac directly, so I think we'll want to include federation resources in the `edit` or `admin` role role to avoid having to maintain explicit rbac rules as the list of federation types grows over time.

Reply:
I believe the issue here is that we are only installing the cluster registry and not making any changes to any existing rbac entity to grant users the ability to use it. `cluster-admin` works as a rolebinding because it grants */*/* permissions.

For now, using cluster-admin in this way is fine. As a follow-up, we should have a more narrowly-scoped role for the federation controller and a role binding to the service account the controller runs under. This is consistent with norms around least-privilege elsewhere in the ecosystem. For example, when you deploy service-catalog, there are service accounts with distinct least-privilege rbac permissions for each process (in that case, apiserver and controller).

Reply:
I don't think namespaced cluster admin is a problem for now.  I'm not clear we'll want to have to maintain rbac roles with explicit type references if the control plane should have access to everything in the namespace, that seems like an operational hassle.

Reply:
No action was required on this original comment - only spurring discussion. I think `cluster-admin` is good for now and we can do a follow-up.